In [1]:
import pandas as pd

<h1>Vaccinations and Countries dataset</h1>

In [2]:
vaccinations_df = pd.read_csv('country_vaccinations.csv')

<h3>Finding the latest available data they have based on the date column</h3>

In [28]:
vaccinations_df.sort_values('date', ascending=False).head(10)

country iso_code        date  total_vaccinations  people_vaccinated  \
4677    Zimbabwe      ZWE  2021-03-01             21456.0            21456.0   
2312       Italy      ITA  2021-03-01           4354008.0          2942345.0   
998   Costa Rica      CRI  2021-03-01            149812.0           103695.0   
2247      Israel      ISR  2021-03-01           8211997.0          4772078.0   
2736    Malaysia      MYS  2021-03-01             25252.0                NaN   
2764    Maldives      MDV  2021-03-01            117115.0                NaN   
1291     Ecuador      ECU  2021-03-01             42114.0            35886.0   
1060     Croatia      HRV  2021-03-01            213068.0           153085.0   
4082    Slovenia      SVN  2021-03-01            174493.0           123087.0   
1678     Germany      DEU  2021-03-01           6394364.0          4234952.0   

      people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
4677                      NaN                  2613.0              2877.0   
2312                1411663.0                 45084.0            101386.0   
998                   46117.0                     NaN              6947.0   
2247                3439919.0                101577.0             94984.0   
2736                      NaN                     NaN              4209.0   
2764                      NaN                  5700.0              5279.0   
1291                   6228.0                     NaN              3183.0   
1060                  59983.0                     NaN              6703.0   
4082                  51406.0                  1032.0              4020.0   
1678                2159412.0                180639.0            161742.0   

      total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
4677                            0.14                           0.14   
2312                            7.20                           4.87   
998                             2.94                           2.04   
2247                           94.88                          55.13   
2736                            0.08                            NaN   
2764                           21.67                            NaN   
1291                            0.24                           0.20   
1060                            5.19                           3.73   
4082                            8.39                           5.92   
1678                            7.63                           5.05   

      people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
4677                                  NaN                           194.0   
2312                                 2.33                          1677.0   
998                                  0.91                          1364.0   
2247                                39.74                         10974.0   
2736                                  NaN                           130.0   
2764                                  NaN                          9766.0   
1291                                 0.04                           180.0   
1060                                 1.46                          1633.0   
4082                                 2.47                          1934.0   
1678                                 2.58                          1930.0   

                                          vaccines  \
4677                             Sinopharm/Beijing   
2312  Moderna, Oxford/AstraZeneca, Pfizer/BioNTech   
998                                Pfizer/BioNTech   
2247                      Moderna, Pfizer/BioNTech   
2736                                       Sinovac   
2764                            Oxford/AstraZeneca   
1291                               Pfizer/BioNTech   
1060                               Pfizer/BioNTech   
4082           Oxford/AstraZeneca, Pfizer/BioNTech   
1678  Moderna, Oxford/AstraZeneca, Pfizer/BioNTech   

                                            source_name

<h3>Counting the number of countries in this dataset</h3>

In [22]:
len(vaccinations_df.country.unique())

118

<h3>Let's see how many countries have record based on the latest available data here</h3>

In [25]:
vaccinations_df[vaccinations_df.date == '2021-03-01']

country iso_code        date  total_vaccinations  \
173              Argentina      ARG  2021-03-01           1049169.0   
188              Australia      AUS  2021-03-01             33702.0   
239                Austria      AUT  2021-03-01            670208.0   
329                Bahrain      BHR  2021-03-01            301103.0   
596                 Brazil      BRA  2021-03-01           8465403.0   
659               Bulgaria      BGR  2021-03-01            223539.0   
729                 Canada      CAN  2021-03-01           1954728.0   
859                  Chile      CHL  2021-03-01           3490572.0   
930               Colombia      COL  2021-03-01            149133.0   
998             Costa Rica      CRI  2021-03-01            149812.0   
1060               Croatia      HRV  2021-03-01            213068.0   
1177               Czechia      CZE  2021-03-01            674040.0   
1291               Ecuador      ECU  2021-03-01             42114.0   
1446               Estonia      EST  2021-03-01            112809.0   
1678               Germany      DEU  2021-03-01           6394364.0   
1792                Greece      GRC  2021-03-01            921189.0   
1911               Hungary      HUN  2021-03-01            974524.0   
1973               Iceland      ISL  2021-03-01             33683.0   
2174           Isle of Man      IMN  2021-03-01             21998.0   
2247                Israel      ISR  2021-03-01           8211997.0   
2312                 Italy      ITA  2021-03-01           4354008.0   
2325                 Japan      JPN  2021-03-01             31785.0   
2534                Latvia      LVA  2021-03-01             64001.0   
2551               Lebanon      LBN  2021-03-01             27154.0   
2657             Lithuania      LTU  2021-03-01            244823.0   
2736              Malaysia      MYS  2021-03-01             25252.0   
2764              Maldives      MDV  2021-03-01            117115.0   
2899                Mexico      MEX  2021-03-01           2526863.0   
3414                Poland      POL  2021-03-01           3383426.0   
3479              Portugal      PRT  2021-03-01            868951.0   
3679                Russia      RUS  2021-03-01           5238966.0   
4017              Slovakia      SVK  2021-03-01            458108.0   
4082              Slovenia      SVN  2021-03-01            174493.0   
4337                Turkey      TUR  2021-03-01           8774718.0   
4373               Ukraine      UKR  2021-03-01              4905.0   
4429  United Arab Emirates      ARE  2021-03-01           6028417.0   
4579         United States      USA  2021-03-01          76899987.0   
4581               Uruguay      URY  2021-03-01             16505.0   
4677              Zimbabwe      ZWE  2021-03-01             21456.0   

      people_vaccinated  people_fully_vaccinated  daily_vaccinations_raw  \
173            755212.0                 293957.0                 24166.0   
188             33702.0                      NaN                  1808.0   
239                 NaN                      NaN                 13538.0   
329            301103.0                      NaN                  2932.0   
596           6542206.0                1923197.0                 31835.0   
659            184764.0                  38775.0                 11094.0   
729                 NaN                      NaN                     NaN   
859           3433987.0                  56585.0                127328.0   
930            149133.0                      NaN                 18555.0   
998            103695.0                  46117.0                     NaN   
1060           153085.0                  59983.0                     NaN   
1177           430724.0                 243316.0                 18852.0   
1291            35886.0                   6228.0                     NaN   
1446            78759.0                  34050.0                     NaN   
1678          4234952.0               

<h3>There are 118 countries with data in this dataset, but only 39 has data on 2021-03-01</h3>

In [27]:
vaccinations_df[vaccinations_df.date == '2021-03-01'].count()

country                                39
iso_code                               39
date                                   39
total_vaccinations                     39
people_vaccinated                      32
people_fully_vaccinated                26
daily_vaccinations_raw                 28
daily_vaccinations                     39
total_vaccinations_per_hundred         39
people_vaccinated_per_hundred          32
people_fully_vaccinated_per_hundred    26
daily_vaccinations_per_million         39
vaccines                               39
source_name                            39
source_website                         39
dtype: int64

<h3>For countries with available data on 2021-03-01, I would like to see total_vaccinations sorted</h3>

In [33]:
latest_vaccinations_df = vaccinations_df[vaccinations_df.date == '2021-03-01']
latest_vaccinations_df.sort_values('total_vaccinations', ascending=False)

country iso_code        date  total_vaccinations  \
4579         United States      USA  2021-03-01          76899987.0   
4337                Turkey      TUR  2021-03-01           8774718.0   
596                 Brazil      BRA  2021-03-01           8465403.0   
2247                Israel      ISR  2021-03-01           8211997.0   
1678               Germany      DEU  2021-03-01           6394364.0   
4429  United Arab Emirates      ARE  2021-03-01           6028417.0   
3679                Russia      RUS  2021-03-01           5238966.0   
2312                 Italy      ITA  2021-03-01           4354008.0   
859                  Chile      CHL  2021-03-01           3490572.0   
3414                Poland      POL  2021-03-01           3383426.0   
2899                Mexico      MEX  2021-03-01           2526863.0   
729                 Canada      CAN  2021-03-01           1954728.0   
173              Argentina      ARG  2021-03-01           1049169.0   
1911               Hungary      HUN  2021-03-01            974524.0   
1792                Greece      GRC  2021-03-01            921189.0   
3479              Portugal      PRT  2021-03-01            868951.0   
1177               Czechia      CZE  2021-03-01            674040.0   
239                Austria      AUT  2021-03-01            670208.0   
4017              Slovakia      SVK  2021-03-01            458108.0   
329                Bahrain      BHR  2021-03-01            301103.0   
2657             Lithuania      LTU  2021-03-01            244823.0   
659               Bulgaria      BGR  2021-03-01            223539.0   
1060               Croatia      HRV  2021-03-01            213068.0   
4082              Slovenia      SVN  2021-03-01            174493.0   
998             Costa Rica      CRI  2021-03-01            149812.0   
930               Colombia      COL  2021-03-01            149133.0   
2764              Maldives      MDV  2021-03-01            117115.0   
1446               Estonia      EST  2021-03-01            112809.0   
2534                Latvia      LVA  2021-03-01             64001.0   
1291               Ecuador      ECU  2021-03-01             42114.0   
188              Australia      AUS  2021-03-01             33702.0   
1973               Iceland      ISL  2021-03-01             33683.0   
2325                 Japan      JPN  2021-03-01             31785.0   
2551               Lebanon      LBN  2021-03-01             27154.0   
2736              Malaysia      MYS  2021-03-01             25252.0   
2174           Isle of Man      IMN  2021-03-01             21998.0   
4677              Zimbabwe      ZWE  2021-03-01             21456.0   
4581               Uruguay      URY  2021-03-01             16505.0   
4373               Ukraine      UKR  2021-03-01              4905.0   

      people_vaccinated  people_fully_vaccinated  daily_vaccinations_raw  \
4579         50732997.0               25466405.0               1663984.0   
4337          6975558.0                1799160.0                226843.0   
596           6542206.0                1923197.0                 31835.0   
2247          4772078.0                3439919.0                101577.0   
1678          4234952.0                2159412.0                180639.0   
4429                NaN                      NaN                  7956.0   
3679          4140332.0                1098634.0                     NaN   
2312          2942345.0                1411663.0                 45084.0   
859           3433987.0                  56585.0                127328.0   
3414          2190526.0                1192900.0                 47072.0   
2899          1959525.0                 567338.0                 71768.0   
729                 NaN                      NaN                     NaN   
173            755212.0                 293957.0                 24166.0   
1911           721677.0                 252847.0                 37586.0   
1792           594053.0               

<h3>
    Going back to the original DataFrame, I want to see how many records are missing values in any of the columns. In this case, we can see that more than 3000 rows are missing data in some columns by checking the shape.
</h3>

In [47]:
vaccinations_df.dropna(how='any').count()

country                                1381
iso_code                               1381
date                                   1381
total_vaccinations                     1381
people_vaccinated                      1381
people_fully_vaccinated                1381
daily_vaccinations_raw                 1381
daily_vaccinations                     1381
total_vaccinations_per_hundred         1381
people_vaccinated_per_hundred          1381
people_fully_vaccinated_per_hundred    1381
daily_vaccinations_per_million         1381
vaccines                               1381
source_name                            1381
source_website                         1381
dtype: int64

In [48]:
vaccinations_df.count()

country                                4678
iso_code                               4366
date                                   4678
total_vaccinations                     3054
people_vaccinated                      2595
people_fully_vaccinated                1746
daily_vaccinations_raw                 2576
daily_vaccinations                     4518
total_vaccinations_per_hundred         3054
people_vaccinated_per_hundred          2595
people_fully_vaccinated_per_hundred    1746
daily_vaccinations_per_million         4518
vaccines                               4678
source_name                            4678
source_website                         4678
dtype: int64

<h3>Checking the total amount of vaccinees given on 2021-03-01</h3>

In [45]:
latest_vaccinations_df['total_vaccinations'].sum()

143348088.0